e2e/deployers/retry.go

```diff
@@ -68,3 +76,88 @@ func WaitWorkloadHealth(ctx types.TestContext, cluster types.Cluster, namespace
 		}
 	}
 }
+
+func WaitForResourceDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	key k8stypes.NamespacedName,
+	obj client.Object,
```

Comment:
key can be create internally, no need to pass it. We can pass the name and the namesapce  as part of the obj.ObjectMeta from the specific type helpers.

We also need a deadline:

    deadline time.Time

So we can wait for deadline when waiting for multiple objects.